app/models/book.rb

```diff
@@ -1,2 +1,9 @@
 class Book < ActiveRecord::Base
+  monetize :price_cents
+
+  
+
+  scope :by_published_date, -> { order(published_date: :desc) } 
+  scope :filter, -> (search) { where("title LIKE ?", "%#{search}%") } 
```

Comment:
Yeah, I don't think you need that scope in there to accomplish what you want. your `.order(sort)` should be doing all the work.